elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceTable.java

```diff
@@ -69,13 +69,16 @@ protected SQLReferenceTable(MetaDataStore metaDataStore, Set<Queryable> queryabl
         this.metaDataStore = metaDataStore;
         this.dictionary = this.metaDataStore.getMetadataDictionary();
 
-        queryables.stream().forEach(queryable -> {
-            Queryable next = queryable;
-            do {
-                resolveAndStoreAllReferencesAndJoins(next);
-                next = next.getSource();
-            } while (next.isNested());
-        });
+        queryables
+           .stream()
+           .filter(queryable -> queryable.isNested() || queryable.isRoot())
```

Comment:
What's this filter for?

Reply:
Wanted to exclude adding those queryable for Dynamic tables which are already there in Static Table, Static Table's queryable already get the preference.
Expression Parser was failing for the scenario: When it is a Query over SQLTable, then Query's ColumnProjection may reference a logical column which is defined in SQLTable but not in Query. It fails at: https://github.com/yahoo/elide/blob/master/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/ExpressionParser.java#L81